integration/benchmark/cmd/ci/main.go

```diff
@@ -198,279 +180,70 @@ func main() {
-					Float64("currentTPS", currentTPS).
-					Int("inflight", inflight).
-					Int("inflightPerWorker", inflightPerWorker).
-					Msg("skipped adjusting TPS")
-
-				lastTxsExecuted = uint(currentStats.TxsExecuted)
-				lastTPS = currentTPS
-				lastTs = nowTs
-
-				continue
-			}
-
-			boundedTPS := boundTPS(unboundedTPS, minTPS, maxTPS)
-			log.Info().
-				Uint("lastTargetTPS", targetTPS).
-				Float64("lastTPS", lastTPS).
-				Float64("currentTPS", currentTPS).
-				Uint("unboundedTPS", unboundedTPS).
-				Uint("targetTPS", boundedTPS).
-				Int("inflight", inflight).
-				Int("inflightPerWorker", inflightPerWorker).
-				Int("txsTimedout", txsTimedout).
-				Msg("adjusting TPS")
-
-			err := lg.SetTPS(boundedTPS)
-			if err != nil {
-				return fmt.Errorf("unable to set tps: %w", err)
-			}
-
-			targetTPS = boundedTPS
-			lastTxsTimedout = currentStats.TxsTimedout
-
-			// SetTPS is a blocking call, so we need to re-fetch the TxExecuted and time.
-			currentStats = workerStatsTracker.GetStats()
-			lastTxsExecuted = uint(currentStats.TxsExecuted)
-			lastTPS = currentTPS
-			lastTs = time.Now()
-		case <-lg.Done():
-			return nil
-		}
-	}
-}
-
-func computeTPS(
-	lastTxs uint,
-	currentTxs uint,
-	lastTs time.Time,
-	nowTs time.Time,
-	lastTPS float64,
-	targetTPS uint,
-	inflight int,
-	maxInflight uint,
-	timedout bool,
-) (bool, float64, uint) {
-	timeDiff := nowTs.Sub(lastTs).Seconds()
-	if timeDiff == 0 {
-		return true, 0, 0
-	}
-
-	currentTPS := float64(currentTxs-lastTxs) / timeDiff
-	unboundedTPS := uint(math.Ceil(currentTPS))
-
-	// If there are timed out transactions we throttle regardless of anything else.
-	// We'll continue to throttle until the timed out transactions are gone.
-	if timedout {
-		return false, currentTPS, uint(float64(targetTPS) * multiplicativeDecrease)
-	}
-
-	// To avoid setting target TPS below current TPS,
-	// we decrease the former one by the multiplicativeDecrease factor.
-	//
-	// This shortcut is only applicable when current inflight is less than maxInflight.
-	if ((float64(unboundedTPS) >= float64(targetTPS)*multiplicativeDecrease) && (inflight < int(maxInflight))) ||
-		(unboundedTPS >= targetTPS) {
-
-		unboundedTPS = targetTPS + additiveIncrease
-	} else {
-		// Do not reduce the target if TPS incresed since the last round.
-		if (currentTPS > float64(lastTPS)) && (inflight < int(maxInflight)) {
-			return true, currentTPS, 0
-		}
-
-		unboundedTPS = uint(float64(targetTPS) * multiplicativeDecrease)
+		log.Fatal().Err(err).Msg("failed to create raw TPS table")
 	}
-	return false, currentTPS, unboundedTPS
-}
 
-func boundTPS(tps, min, max uint) uint {
-	switch {
-	case tps < min:
-		return min
-	case tps > max:
-		return max
-	default:
-		return tps
+	log.Info().Msg("Uploading data to BigQuery")
+	err = db.saveRawResults(
+		ctx,
+		*bigQueryDatasetFlag,
+		*bigQueryRawTableFlag,
+		recorder.BenchmarkResults,
+		*repoInfo,
+		BenchmarkInfo{BenchmarkType: loadType},
+		defaultEnvironment(),
```

Comment:
Should the environment be settable? Do we need to add support for other environments to pass to `main` as a flag?

Reply:
for now, environment is the property of the linux host (kernel version and hostname). -- we'll eventually need to pass a custom one, but that can be done later.